src/conduits/RwaMultiSwapOutputConduit.sol

```diff
@@ -0,0 +1,461 @@
+        emit Pick(who);
+    }
+
+    /**
+     * @notice Sets `psm` address which will be used for swap DAI -> GEM.
+     * @param _psm PSM address.
+     * @dev `psm` address should have been whitelisted using `clap`.
+     */
+    function hook(address _psm) external {
+        require(can[msg.sender] == 1, "RwaMultiSwapOutputConduit/not-operator");
+        require(pal[_psm] == 1 || _psm == address(0), "RwaMultiSwapOutputConduit/not-pal");
+
+        psm = _psm;
+
+        emit Hook(_psm);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Swaps the DAI balance of this contract into GEM through the PSM and push it into the recipient address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function push() external onlyMate {
+        _doPush(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param wad DAI amount.
+     */
+    function push(uint256 wad) external onlyMate {
+        _doPush(wad);
+    }
+
+    /**
+     * @notice Flushes out any DAI balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external onlyMate {
+        _doQuit(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out the specified amount of DAI to the `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param wad DAI amount.
+     */
+    function quit(uint256 wad) external onlyMate {
+        _doQuit(wad);
+    }
+
+    /**
+     * @notice Flushes out `amt` of `token` sitting in this contract to `usr` address.
+     * @dev Can only be called by the admin.
+     * @param token Token address.
+     * @param usr Destination address.
+     * @param amt Token amount.
+     */
+    function yank(
+        address token,
+        address usr,
+        uint256 amt
+    ) external auth {
+        GemAbstract(token).transfer(usr, amt);
+        emit Yank(token, usr, amt);
+    }
+
+    /**
+     * @notice Return Gem address of the selected PSM
+     * @return gem Gem address.
+     */
+    function gem() external view returns (address) {
+        return psm != address(0) ? GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem() : address(0);
+    }
+
+    /**
+     * @notice Calculates the amount of GEM received for swapping `wad` of DAI.
+     * @param wad DAI amount.
+     * @return amt Expected GEM amount.
+     */
+    function expectedGemAmt(uint256 wad) public view returns (uint256 amt) {
+        if (psm == address(0)) return 0;
+
+        uint256 decimals = GemAbstract(GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem()).decimals();
+        uint256 to18ConversionFactor = 10**_sub(18, decimals);
```

Comment:
Because of [this check](https://github.com/makerdao/rwa-toolkit/pull/31/files#diff-8eb928d7ad46715b8ed860cc35c93936921dbd7d19d22b467995776ac51a1a19R253-R256), we can use regular subtraction here:

```suggestion
        uint256 to18ConversionFactor = 10**(18 - decimals);
```